gateway-js/src/utilities/cleanErrorOfInaccessibleNames.ts

```diff
@@ -0,0 +1,69 @@
+import { GraphQLError, GraphQLSchema } from 'graphql';
+
+export function cleanErrorOfInaccessibleNames(
+  schema: GraphQLSchema,
+  error: GraphQLError,
+): GraphQLError {
+  const typeRegex = /"([_A-Za-z][_0-9A-Za-z]*)"/g;
+  const typeDotFieldRegex =
+    /"([_A-Za-z][_0-9A-Za-z]*)\.([_A-Za-z][_0-9A-Za-z]*)"/g;
+
+  const schemaTypes = Object.keys(schema.getTypeMap());
+  let match: RegExpExecArray | null;
+  const typeDotFieldMatches: [string, string][] = [];
+  while ((match = typeDotFieldRegex.exec(error.message)) !== null) {
+    typeDotFieldMatches.push([match[1], match[2]]);
+  }
+
+  if (typeDotFieldMatches && typeDotFieldMatches.length > 0) {
+    const inaccessibleFields = typeDotFieldMatches.filter(
+      ([typeName, fieldName]) => {
+        const type = schema.getType(typeName);
+        if (!type) {
+          return true;
+        } else {
+          const field =
+            'getFields' in type ? type.getFields()[fieldName] : null;
+          return !field;
+        }
+      },
+    );
+    if (inaccessibleFields.length > 0) {
+      error.message = error.message.replace(
+        new RegExp(
+          inaccessibleFields
+            .map(([type, field]) => `("${type}.${field}")`)
+            .join('|'),
+          'g',
+        ),
+        '[inaccessible field]',
+      );
+    }
+  }
+
+  const typeMatches: string[] = [];
+  while ((match = typeRegex.exec(error.message)) !== null) {
+    if (match[1] !== 'resolveType' && match[1] !== 'isTypeOf') {
+      typeMatches.push(match[1]);
+    }
+  }
+  if (typeMatches && typeMatches.length > 0) {
+    const inaccessibleTypes = typeMatches.filter(
+      (match) => !schemaTypes.includes(match),
```

Comment:
Since `schemaTypes.includes()` in effect performances a linear search on `Object.keys(schema.getTypeMap())`, wouldn't it be easier and more performant to use `schema.getType()` directly?

Reply:
I did that in the fields names bit, not sure why I didn't do it in both places! 👍